src/main/scala/org/monarchinitiative/dosdp/DOSDP.scala

```diff
@@ -89,19 +88,22 @@ trait PrintfText {
 
   def annotations: Option[List[Annotations]]
 
-  def replaced(bindings: Option[Map[String, SingleValue]]): Option[String] = PrintfText.replaced(this.text, this.vars, None, bindings, shouldQuote)
+  def multi_clause: Option[MultiClausePrintf]
+
+  def replaced(bindings: Option[Map[String, Binding]]): Option[String] = PrintfText.replaced(this.text, this.vars, this.multi_clause, bindings, shouldQuote)
 
   def shouldQuote: Boolean
 
 }
 
 object PrintfText {
 
-  def replaced(text: Option[String], vars: Option[List[String]], multi_clause: Option[MultiClausePrintf], bindings: Option[Map[String, SingleValue]], quote: Boolean): Option[String] = {
-    val replacedText = text.flatMap(content => replaceText(content, vars, bindings, quote))
+  def replaced(text: Option[String], vars: Option[List[String]], multi_clause: Option[MultiClausePrintf], bindings: Option[Map[String, Binding]], quote: Boolean): Option[String] = {
+    val singleValueBindings = bindings.map( x => x.collect { case (key, SingleValue(value)) => (key, SingleValue(value)) })
+    val replacedText = text.flatMap(content => replaceText(content, vars, singleValueBindings, quote))
     val replacedClause = multi_clause.flatMap(content => replaceMultiClause(content, bindings, quote))
     val replacedPrintf = replacedText ++ replacedClause
-    if (replacedPrintf.isEmpty) None else replacedPrintf.headOption
+    replacedPrintf.headOption
```

Comment:
I think here you want to return the first of the Options that is non-empty, or else None? In that case can you replace these three lines with `replacedText.orElse(replacedClause)`?

Reply:
I made the change.

Reply:
You are right, thank you. Tests are still passing and it is more elegant now